zfsbootmenu/lib/zfsbootmenu-ui.sh

```diff
@@ -623,11 +616,35 @@ populate_be_list() {
       zwarn "canmount=on set for '${fs}', should be canmount=noauto"
     fi
 
-    candidates+=( "${fs}" )
-  done <<< "$(zfs list -H -o name,canmount,mountpoint,org.zfsbootmenu:active -S "${sort_key}")"
+    echo "${fs}"
+    ret=0
+  done <<< "$(zfs list -H -o "${list_fields}" -S "${sort_key}")"
+
+  # put bootfs at the end
+  [ -n "${BOOTFS}" ] && [ -n "${have_bootfs}" ] && echo "${BOOTFS}"
+  return "${ret}"
+}
+
+# arg1: path to BE list
+# prints: nothing
+# returns: 0 iff at least one valid BE was found
+
+populate_be_list() {
+  local be_list fs candidates ret
+
+  be_list="${1}"
+  if [ -z "${be_list}" ]; then
+    zerror "be_list is undefined"
+    return 1
+  fi
+  zdebug "be_list set to ${be_list}"
+
+  sort_key="$( get_sort_key )"
```

Comment:
I don't think this needs to get a sort key any more

Reply:
Done